test/unit/parser_test.rb

```diff
@@ -1180,6 +1180,13 @@ def test_cast_with_invalid_type
     assert_equal '0', expressions.first[:value]
   end
 
+  test 'Arithmetic grouping should not fail on nesting' do
+    filter = "BathroomsTotalDecimal Eq ((3.4 add 2.6) mul (1 add 1))"
+    @parser = Parser.new
+    @parser.parse(filter)
+    assert !@parser.errors?
```

Comment:
This would fail before? Wow.

I'm not sure if there's another spot that tests this, but you may also want to go a step further and assert that the nesting worked correctly too.